docs/docsite/rst/dev_guide/collection_requirements.rst

```diff
@@ -0,0 +1,437 @@
+
+    - Dropping support for Python 2.6 in the other environment means that you are dropping support for RHEL6.  RHEL6 ended full support in November, 2020, but some users are still using RHEL6 under extended support contracts (ELS) until 2024.  ELS is not full support; not all CVEs of the python-2.6 interpreter are fixed, for instance.
+
+    - Dropping support for Python 3.5 means that Python 2.7 has to be installed on Ubuntu Xenial (16.04) and that you have to support Python 2.7.
+
+    Also, note that dropping support for a Python version for an existing module/plugin is a breaking change, and thus requires a major release. Hence, a collection MUST announce dropping support for Python versions in their changelog, if possible in advance (for example, in previous versions before support is dropped).
+
+Documentation
+~~~~~~~~~~~~~
+
+* If everything in your collection supports the same Python versions as the collection-supported versions of ansible-core, you do not need to document Python versions.
+* If your collection does not support those Python versions, you MUST document which versions it supports in the README.
+* If most of your collection supports the same Python versions as ansible-core, but some modules and plugins do not, you MUST include the supported Python versions in the documentation for those modules and plugins.
+
+For example, if your collection supports Ansible 2.9 to ansible-core 2.13, the Python versions supported for modules are 2.6, 2.7, and 3.5 and newer (until at least 3.10), while the Python versions supported for plugins are 2.7 and 3.5 and newer (until at least 3.10). So if the modules in your collection do not support Python 2.6, you have to document this in the README, for example ``The content in this collection supports Python 2.7, Python 3.5 and newer.``.
+
+Standards for developing module and plugin utilities
+====================================================
+
+* ``module_utils`` and ``plugin_utils`` can be marked for only internal use in the collection, but they MUST document this and MUST use a leading underscore for filenames.
+* It is a breaking change when you make an existing ``module_utils`` private and in that case the collection requires a major version bump.
+* Below are some recommendations for ``module_utils`` documentation: 
+
+  * No docstring: everything we recommend for ``other-environment`` is supported.
+  * The docstring ``'Python versions supported: same as for controller-environment'``: everything we recommend for ``controller-environment`` is supported.
+  * The docstring with specific versions otherwise: ``'Python versions supported: '``.
+
+Repo structure
+===============
+
+galaxy.yml
+----------
+
+* The ``tags`` field MUST be set.
+* Collection dependencies must meet a set of rules. See the section on `Collection Dependencies <collection_dependencies_>`_ for this.
```

Comment:
```suggestion
* Collection dependencies must meet a set of rules. See  :ref:`collection_dependencies_requirements` for details.
```

Reply:
Two items fixed here:
1 - because of the duplicate label problem, we neede to change the label based on the change at https://github.com/ansible/ansible/pull/78637/files#diff-213450fadf3303b3003c308ba994d71c799769440a5c384726e9150d2501f51bR378 below.

2 - this is an internal link so we want to use the :ref: format.